libres/lib/enkf/obs_data.cpp

```diff
@@ -450,9 +445,6 @@ matrix_type *obs_data_allocR(const obs_data_type *obs_data) {
             obs_block_initR(obs_block, R, &obs_offset);
         }
     }
-
-    matrix_set_name(R, "R");
-    matrix_assert_finite(R);
```

Comment:
By doing this we assume that an error will be thrown if we come in a bad state? I am now assuming this was added for a reason at some point.

Reply:
> Regardless, it's a silly check, and is quite expensive considering you have to check every cell for something that you should assume is true.

I guess silly is in the mind of beholder - but as you indicate yourself with the `E` matrix example the probability of `nan/inf` creeping into these matrices is distinctly non zero, although the bug in that case would be with the *input* to the matrix constructor and not with the matrix itself.

Reply:
_Was_ distinctly non-zero. We have fixed the problem some time ago.

Reply:
Well - these assert have caught numerous problems in the past; but they have probably outlived their usefullness - I agree this is good time to remove them.